### PR TITLE
Make errors more informative and friendly

### PR DESCRIPTION
**Pull request by smothers**
Errors were CLJS and datascript specific which does not help JS React devs. This PR reformats the most common errors so they make sense with the homebase-react JS API. It also attempts to be more educational, providing examples and in one case generating code to resolve the error.

### CODE REVIEW

File: src/homebase/js.cljs
Comment (smothers) [inline src/homebase/js.cljs:202]: @rubinovitz you might already know this, but if you ever want to work on testing these comments are a good place to start. I use inline comments while developing in the repl. Then when it all works I like to cut my comments out and paste them into my test suite to use as regression tests. The code in these comments needs to be massaged into boolean assertions, but otherwise they should represent a pretty solid amount of coverage.

**Comment by rubinovitz** [comment]
@smothers do you know the "test" it's expecting to run here that is required but doesn't seem to be running?

**Comment by smothers** [comment]
> @smothers do you know the "test" it's expecting to run here that is required but doesn't seem to be running?

@rubinovitz I do. It's not going to work until the CI PR is merged and then synced with this branch, that's why I want to merge the CI PR first.

**Comment by rubinovitz** [comment]
@smothers got it thanks! I want to get the Benchmarks in first and then I will approve CI.

**Comment by smothers** [comment]
> @smothers got it thanks! I want to get the Benchmarks in first and then I will approve CI.

@rubinovitz If possible can this be decoupled from the work on benchmarks? That branch still needs updates to be made and another review.

**Comment by rubinovitz** [comment]
@smothers I’m trying to avoid having to do another round of updates on that after this is merged in. I’m doing the benchmarks updates today.

**Review by rubinovitz — DISMISSED** [review]

**Review by rubinovitz — APPROVED** [review]
Nice

**Pull request merged**
